flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -659,12 +658,14 @@ else if (!props.forceCheckpoint()) {
 					cancellerHandle.cancel(false);
 				}
 
-				// trigger the master hooks for the checkpoint
-				final Map<String, MasterState> masterStates = MasterHooks.triggerMasterHooks(masterHooks.values(),
-						checkpointID, timestamp, executor, Time.milliseconds(checkpointTimeout));
-				for (Map.Entry<String, MasterState> entry : masterStates.entrySet()) {
-					checkpoint.acknowledgeMasterState(entry.getKey(), entry.getValue());
+				// TODO, asynchronously snapshots master hook without waiting here
```

Comment:
Please also elaborate in the commit message of `[FLINK-14344][checkpointing] MasterHooks supports asynchronously snapshotting state`. For example:

```
[FLINK-14344][checkpointing] MasterHooks class supports asynchronous triggering

This commits add supports for asynchronous triggering `MasterHooks`.
As for now, they are still being fired synchronously by the `CheckpointCoordinator`.
```